wavedump/src/analyze-waveforms

```diff
@@ -90,134 +83,284 @@ def get_bins(x):
     """
     
     bin_width = 0.5*iqr(x)/(len(x)**(1/3.0))
+    if bin_width == 0:
+        print('Zero bin width! Making it 0.0001.')
+        bin_width = 0.0001
     return np.arange(np.min(x),np.max(x),bin_width)
 
+def chunks(lst, n):
+    """Yield successive n-sized chunks from lst."""
+    for i in range(0, len(lst), n):
+        yield (i,i + n)
+
+def convert_data(f, channel, start, stop):
+    """
+    Reads data from opened hdf5 file `f`.
+    Gets the events from `start` to `stop` in the dataset `channel`.
+    """    
+    if 'data_source' in list(f.attrs):
+        if f.attrs['data_source'] == b'CAEN':
+            
+            xinc = 1/(f.attrs['drs4_frequency'] * 10**6)
+            points = f.attrs['record_length']
+            x = np.linspace(0, xinc * points, int(points)) - xinc * points * (1 - f.attrs['post_trigger']/100)
+
+            y = f[channel][start:stop] * (1/2**12)
+
+    elif 'yinc' in dict(f[channel].attrs):
+        x = f[channel].attrs['xorg'] + np.linspace(0,f[channel].attrs['xinc']*f[channel].attrs['points'],int(f[channel].attrs['points']))
+
+        if True: # ':WAVeform:FORMat' in dict(f['settings'].attrs) and f['settings'].attrs[':WAVeform:FORMat'] != 'ASC':
```

Comment:
I've now forgotten why we changed this. I remember it was some bug with the way things were stored in the hdf5 file, maybe you can add a FIXME comment above, so I remember to go figure out what the problem was and fix it.